### DISCUSSION TITLE
Fast-DDS does not support modern DDS C++ API

### DISCUSSION THREAD

**Discussion by trittsv** [eProsima/Fast-DDS#4965] [Ideas]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

```
dds::domain::DomainParticipant participant(0);

   
        dds::topic::Topic<HelloWorldData::Msg> topic(participant, "HelloWorldData_Msg");

        dds::pub::Publisher publisher(participant);

  
        dds::pub::DataWriter<HelloWorldData::Msg> writer(publisher, topic);

        while (writer.publication_matched_status().current_count() == 0) {
            std::this_thread::sleep_for(std::chrono::milliseconds(20));
        }

     
        HelloWorldData::Msg msg(1, "Hello World");
        writer.write(msg);

```

### Current behavior

Is not compileable.

### Steps to reproduce

Try to compile the provided example 

### Fast DDS version/commit

2.14.0

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Default configuration, UDPv4 & SHM

### Additional context

API is incomplete to be compatible to modern dds cpp api.

**Comment by EduPonz** [comment]
I've converted this into an "ideas" discussion, as it is a wish and definitely not an issue with the current Fast DDS

**Comment by trittsv** [comment]
There is now a non-official modern c++ api wrapper: https://github.com/trittsv/Fast-DDS-modern-cpp-api
